src/engine/source/api/src/catalog/catalog.cpp

```diff
@@ -258,6 +274,17 @@ std::optional<base::Error> Catalog::putResource(const Resource& item, const std:
             fmt::format("Invalid resource type '{}' for PUT operation", Resource::typeToStr(item.m_type))};
     }
 
+    Format contentFormat;
+    try
+    {
+        auto tmp = json::Json {content.c_str()};
+        contentFormat = Format::json;
+    }
+    catch (const std::exception& e)
+    {
+        contentFormat = Format::yaml;
+    }
```

Comment:
How can you be sure that if it is not json it is a valid yaml? You shouldn't auto deduce the type, it's information you already have.